source/kraken/realtime.cpp

```diff
@@ -46,38 +46,15 @@ namespace navitia {
 
 namespace nd = type::disruption;
 
-static bool is_handleable(const transit_realtime::TripUpdate& trip_update){
-    namespace bpt = boost::posix_time;
-
-    auto log = log4cplus::Logger::getInstance("realtime");
-    // Case 1: the trip is cancelled
-    if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_CANCELED) {
-        return true;
-    }
-    // Case 2: the trip is not cancelled, but modified (ex: the train's arrival is delayed or the train's
-    // departure is brought forward), we check the size of stop_time_update because we can't find a proper
-    // enum in gtfs-rt proto to express this idea
-    if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_SCHEDULED
-            && trip_update.stop_time_update_size()) {
-
-        // WARNING: here trip.start_date is considered UTC, not local
-        //(this date differs if vj starts during the period between midnight UTC and local midnight)
-        auto start_date = boost::gregorian::from_undelimited_string(trip_update.trip().start_date());
-
-        auto start_first_day_of_impact = bpt::ptime(start_date, bpt::time_duration(0, 0, 0, 0));
-        for (const auto& st: trip_update.stop_time_update()) {
-            auto ptime_arrival = bpt::from_time_t(st.arrival().time());
-            auto ptime_departure = bpt::from_time_t(st.departure().time());
-            if (ptime_arrival < start_first_day_of_impact
-                    || ptime_departure < start_first_day_of_impact) {
-                LOG4CPLUS_WARN(log, "Trip Update " << trip_update.trip().trip_id() << ": Stop time "
-                                    << st.stop_id() << " is before the day of impact");
-                return false;
-            }
-        }
+static bool is_circulating(const transit_realtime::TripUpdate& trip_update)
+{
+    if ((trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_SCHEDULED ||
+         trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_ADDED)
+         && trip_update.stop_time_update_size()) {
```

Comment:
```suggestion
         && trip_update.stop_time_update_size() > 1) {
```
If there is only one `stop_time_update` we can discard it

Reply:
rather that ?
```
 && trip_update.stop_time_update_size() >= 1) {
```
if the GTFS-RT contains one stop_time we can process ?

Reply:
@kinnou02 you mean that we are supposed to get trips that are complete, and a trip with only one stop is useless, so we should get rid of it, I guess.
So he means `> 1`.

Do we do that when we read NTFS?
This would remove the disruption also (not sure it's bad though).
We could as well reject them in Kirin.
Both are fine to me.

Reply:
A vj with only one stop isn't a vj, it's garbage :)

Reply:
I am all right with that. I take your suggest